Cython/Compiler/PyrexTypes.py

```diff
@@ -3130,7 +3130,9 @@ def declaration_code(self, entity_code,
             arg_decl_code = "void"
         trailer = ""
         if (pyrex or for_display) and not self.return_type.is_pyobject:
-            if self.exception_value and self.exception_check:
+            if not self.exception_check and not self.exception_value:
+                trailer = " noexcept"
+            elif self.exception_value and self.exception_check:
                 trailer = " except? %s" % self.exception_value
             elif self.exception_value:
                 trailer = " except %s" % self.exception_value
```

Comment:
I think changing the order of the if-condition would help the flow of reading this construct (keep `exception_value` first for consistency, step through all 4 combinations)
```suggestion
            if self.exception_value and self.exception_check:
                trailer = " except? %s" % self.exception_value
            elif self.exception_value and not self.exception_check:
                trailer = " except %s" % self.exception_value
            elif not self.exception_value and self.exception_check:
                trailer = " except +"
            elif not self.exception_value and not self.exception_check:
                trailer = " noexcept"
```